src/elements/content-sidebar/versions/VersionsItem.js

```diff
@@ -104,10 +108,10 @@ const VersionsItem = ({
 
     // Version action helpers
     const canPreview = can_preview && !isDeleted && !isLimited && !isRestricted;
-    const showDelete = can_delete && !isDeleted && !isCurrent;
+    const showDelete = can_delete && !shouldDisableVersionChanges && !isDeleted && !isCurrent;
     const showDownload = can_download && !isDeleted && isDownloadable;
-    const showPromote = can_upload && !isDeleted && !isCurrent;
-    const showRestore = can_delete && isDeleted;
+    const showPromote = can_upload && !shouldDisableVersionChanges && !isDeleted && !isCurrent;
+    const showRestore = can_delete && !shouldDisableVersionChanges && isDeleted;
```

Comment:
We should use permissions to control whether a user has the ability to do something. That way, our API will enforce the same restrictions.